build_all.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -e
+
+for d in */ ; do
```

Comment:
Contracts has to be built in the specific order. Especially, the factory contract has to be built after staking pool, because it hardcodes the code inside.